src/query/atomWithInfiniteQuery.ts

```diff
@@ -24,6 +24,8 @@ export type AtomWithInfiniteQueryOptions<
   queryKey: QueryKey
 }
 
+const getInitialDataCache = new WeakMap()
```

Comment:
I noticed that `getInitialData` will run many times when an atom is used -- this seems to not be desirable. @dai-shi curious what you think about this method.

Reply:
At a glance, it feels weird. Do you know why `getinitialData` runs many times? I expect there should be a better fix.

Reply:
I'm not sure -- but when I ran a single test, and did `console.log('running')` in `getInitialData`, it ran 4 times. This could be a test-only phenomena though. I'll test in an app

Reply:
this was also to fix the tests
see this commit failing:
https://github.com/pmndrs/jotai/pull/653/commits/b1a931530cbd19215bd25ec945131ecdac48577e
and this passes when ensuring `getInitialData` runs only once per atom
https://github.com/pmndrs/jotai/pull/653/commits/8d3f49ae08e33268c23d35503f9488b69852994b

Reply:
OK, if it's test failing, it's easier for me to look into it.

> This could be a test-only phenomena though. I'll test in an app

Let me know how it goes, and if I can jump in to investigate.

Reply:
This does seem to _only_ happen in the testing context. I'm curious if it would still render the same atom multiple times if we used the hooks testing library instead.

Reply:
If you'd like to look, that'd be helpful. The issue is the function runs 4 times very quickly, and as the queryClient updates over time, the 3rd time it runs, it receives a value thus never suspends (or suspends very very quickly).

Reply:
This was a mistake, feel free to ignore :)